File: loja-python/tests/test_carrinho.py
from loja import Produto, Carrinho

def test_carrinho_total():
    c = Carrinho()

    c.adicionar(Produto("Livro", 50.0))
    c.adicionar(Produto("Caneta", 5.0))

    assert c.quantidade() == 2
    assert c.total() == 55.0

File: loja-cpp/tests/test_carrinho.cpp
#include "produto.hpp"
#include "carrinho.hpp"

#include <cassert>

int main()
{
    Carrinho carrinho;

    carrinho.adicionar(Produto("Livro", 50.0));
    carrinho.adicionar(Produto("Caneta", 5.0));

    assert(carrinho.quantidade() == 2);
    assert(carrinho.total() == 55.0);

    return 0;
}
